phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -465,7 +465,14 @@ private static String toString(List<byte[]> gps) {
     }
 
     private List<List<Scan>> getParallelScans() throws SQLException {
-        return getParallelScans(EMPTY_BYTE_ARRAY, EMPTY_BYTE_ARRAY);
+        if (scan == null
+                || (ScanUtil.isLocalIndex(scan)
+                        && Bytes.compareTo(context.getScan().getStartRow(), scan.getStartRow()) == 0 && Bytes
+                        .compareTo(context.getScan().getStopRow(), scan.getStopRow()) == 0)) {
+            return getParallelScans(EMPTY_BYTE_ARRAY, EMPTY_BYTE_ARRAY);
```

Comment:
This check detects whether scan boundaries  are equal to context scan boundaries or not. If they are same we are going by getting all parallel scans for the table. Will document it.